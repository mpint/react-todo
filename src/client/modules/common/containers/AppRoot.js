import React, { PropTypes, Component } from 'react';
import { revisionStyles } from '~/styles/inline';
import { setLocalStorageItem } from '~/modules/common/state/localStorage';
import { version, commit } from '../../../../../.revision';
import MenuList from '~/modules/common/components/MenuList';

class AppRoot extends Component {
  constructor() {
    super();

    this.state = {
      viewportWidth: window.innerWidth,
      viewportHeight: window.outerHeight,
      contentWidth: window.innerWidth,
      contentHeight: window.outerHeight
    };
  }

  componentDidUpdate() {
    const currentPath = this.props.location.pathname;
    setLocalStorageItem('app-history', currentPath);
  }

  componentDidMount = () => {
    window.addEventListener('resize', this.handleResize);
  }

  componentWillUnmount = () => {
    window.removeEventListener('resize', this.handleResize);
  }

  getChildContext = () =>  {
    return {
      viewportWidth: this.state.viewportWidth,
      viewportHeight: this.state.viewportHeight,
      contentWidth: this.state.contentWidth,
      contentHeight: this.state.contentHeight
    };
  }

  handleResize = () => {
    this.setState({
      viewportWidth: window.innerWidth,
      viewportHeight: window.outerHeight,
      contentWidth: window.innerWidth,
      contentHeight: window.outerHeight
    });
  }

  render () {
    return (
      <div>
        <div className="menu-container">
          <MenuList
            pageConfig={ this.context.config.pages } />
        </div>

        <div className="grid-container">
          { this.props.children }
        </div>

        <div style={ revisionStyles }>
          { version && `ver. ${version}` }
          { commit && ` (${commit.slice(0, 7)})`}
        </div>
      </div>
    );
  }
}

AppRoot.contextTypes = {
  muiTheme: PropTypes.object.isRequired,
  config: PropTypes.object.isRequired
};

AppRoot.childContextTypes = {
  viewportHeight: PropTypes.number.isRequired,
  viewportWidth: PropTypes.number.isRequired,
  contentWidth: PropTypes.number.isRequired,
  contentHeight: PropTypes.number.isRequired
};

export default AppRoot;
