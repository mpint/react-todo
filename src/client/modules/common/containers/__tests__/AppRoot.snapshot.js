import React from 'react';
import renderer from 'react-test-renderer';

import { wrapWithContext, store, muiTheme, config } from '~/lib/test.helpers';

import AppRoot from '../AppRoot';

describe('AppRoot', function() {
  const contextTypes = { muiTheme: React.PropTypes.object, config: React.PropTypes.object };

  it('should render', function() {
    const component = renderer.create(
      wrapWithContext(
        { muiTheme, config },
        contextTypes,
        <AppRoot />
      )
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
