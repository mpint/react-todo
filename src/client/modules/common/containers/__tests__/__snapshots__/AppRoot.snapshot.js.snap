exports[`AppRoot should render 1`] = `
<div>
  <div
    className="menu-container">
    <ul>
      <a
        onClick={[Function]}
        style={Object {}}>
         
        Home
         
      </a>
      <span>
        |
      </span>
      <a
        onClick={[Function]}
        style={Object {}}>
         
        Interview
         
      </a>
      <span>
        |
      </span>
      <a
        onClick={[Function]}
        style={Object {}}>
         
        Credits
         
      </a>
    </ul>
  </div>
  <div
    className="grid-container" />
  <div
    style={
      Object {
        "bottom": 10,
        "color": "white",
        "fontSize": 10,
        "left": 10,
        "position": "fixed",
        "zIndex": 1500,
      }
    }>
    ver. 0.0.0
     (c8d512b)
  </div>
</div>
`;
